include/seqan3/alphabet/concept.hpp

```diff
@@ -47,42 +47,12 @@ namespace seqan3
 {
 
 // ------------------------------------------------------------------
-// free functions to member function wrapper
-// ------------------------------------------------------------------
-// move to alphabet_detail.hpp?
-
-
-/* The public alphabet concept requires only free function access
- * for type that have member functions we create a wrapper here
- * so you don't have to repeat it.
- */
-//! \privatesection
-namespace detail
-{
-
-template <typename t>
-concept bool internal_alphabet_concept = requires (t t1)
-{
-    typename t::char_type;
-    typename t::integral_type;
-    t::value_size;
-
-    { t1.to_char() } -> typename t::char_type;
-    { t1.to_integral() } -> typename t::integral_type;
-
-    { t1.from_char('a')   } -> t;
-    { t1.from_integral(0) } -> t;
-};
-} // namespace seqan3::detail
-//! \publicsection
-
-// ------------------------------------------------------------------
 // type metafunctions operator
 // ------------------------------------------------------------------
 
 //! Type metafunction that returns the `char_type` defined inside an alphabet type.
 template <typename alphabet_type>
-    requires detail::internal_alphabet_concept<alphabet_type>
+    requires requires (alphabet_type c) { typename alphabet_type::char_type; }
```

Comment:
There is no real duplication, every function now specifies it's own requirements.